qt/widgets/common/src/AlgorithmProgress/AlgorithmProgressDialogWidget.cpp

```diff
@@ -54,16 +53,21 @@ AlgorithmProgressDialogWidget::AlgorithmProgressDialogWidget(
 }
 
 std::pair<QTreeWidgetItem *, QProgressBar *>
-AlgorithmProgressDialogWidget::addAlgorithm(
-    std::string name, std::vector<Mantid::Kernel::Property *> properties) {
+AlgorithmProgressDialogWidget::addAlgorithm(Mantid::API::IAlgorithm_sptr alg) {
+  const auto name = alg->name();
+  const auto properties = alg->getProperties();
   const auto item =
       new QTreeWidgetItem(m_tree, QStringList{QString::fromStdString(name)});
 
   m_tree->addTopLevelItem(item);
   auto progressBar = new QProgressBar(m_tree);
   progressBar->setAlignment(Qt::AlignHCenter);
 
+  auto cancelButton = new AlgorithmProgressDialogWidgetCancelButton(
+      alg, QString::fromStdString("Cancel"), m_tree);
```

Comment:
Would it be better to just hard code "Cancel" in the class? This button isn't used anywhere else. That way the parsing is avoided on every addAlgortihm call.

Reply:
Makes sense :+1: